src/clarity-angular/datagrid/datagrid.ts

```diff
@@ -223,9 +227,18 @@ export class Datagrid implements AfterContentInit, AfterViewInit, OnDestroy {
     ngAfterViewInit() {
         // TODO: determine if we can get rid of provider wiring in view init so that subscriptions can be done earlier
         this.triggerRefresh();
-        this._subscriptions.push(this.sort.change.subscribe(() => this.triggerRefresh()));
-        this._subscriptions.push(this.filters.change.subscribe(() => this.triggerRefresh()));
-        this._subscriptions.push(this.page.change.subscribe(() => this.triggerRefresh()));
+
+        // these changes (sort, filters, page) all should call triggerRefresh. The triggerRefresh
+        // function takes a snapshot of the most recent state and emits the refresh event. Here, we combine
+        // and debounce them so that the rapid successive changes don't throw the infamous
+        // "expression has changed after it was checked" error.
+        let merged = Observable.merge(this.sort.change, this.filters.change, this.page.change)
+            .debounce(() => Observable.interval(300));
```

Comment:
300ms debounce seems gigantic. As I mentioned on Slack, I'd much rather not re-emit duplicate states than do a debounce. And I don't believe it would be complicated to implement: you still combine the observables, turn them into an observable of `State` objects, and remove duplicates. Or something like this. You can even manually just test if the state has changed in the `triggerRefresh()` method, and abort if it hasn't. Many possible solutions.

But if really _really_ we have to debounce, then at least put something that the user can't actually see, like 30ms.